docs/reference/contributing/target/ResetReason.md

```diff
@@ -0,0 +1,38 @@
+### Reset Reason
+
+The Reset Reason API provides a platform-independent method of checking the cause of the last system reset. Implementing the Reset Reason API is not mandatory, but by implementing it, you can check for erroneous system resets at boot time.
+
+#### Assumptions
+
+##### Defined behavior
+
+- The function `hal_reset_reason_clear` clears the reset reason registers for the next system boot.
+- The function `hal_reset_reason_get` will get the reset reason currently in the reset reason registers. This may not necessarily be the value of the reset reason registers that were set at boot time if they have been cleared since.
```

Comment:
I'm not sure what this second bullet is saying. Please clarify.

Reply:
I guess it's somewhat superfluous. The reset reason API is meant to get the reason for the reset at boot time. By the time the function is called by the user to read the value it may have been cleared by some other part of the application. So, whilst there may have been a reset reason in the registers when the system was started, it may not be available when the user comes to check it.